### PR TITLE
Add typescript template

### PR DESCRIPTION
**Pull request by CalebCurry**
## Motivation

Typescript good

## Test Plan

(Write your test plan here)

## Related PRs

(Link your related PRs here)

### CODE REVIEW

File: create-aleo-app/src/index.ts [hunk 1]
```diff
@@ -49,6 +49,17 @@ const FRAMEWORKS: Framework[] = [
         color: lightGreen,
       },
     ],
+  },{
+    name: "react-typescript",
```
Comment (onetrickwolf) [inline create-aleo-app/src/index.ts]: This should probably be a variant under React not it's own thing. Would take a look at how vite does it: https://github.com/vitejs/vite/blob/main/packages/create-vite/src/index.ts#L90

File: create-aleo-app/template-react-typescript/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "build": "webpack",
+    "dev-webpack": "webpack serve",
+    "build:vite": "vite build",
+    "lint": "eslint . --ext js,jsx --report-unused-disable-directives --max-warnings 0",
+    "preview": "vite preview",
+    "install-leo": "./install.sh"
+  },
+  "dependencies": {
+    "@aleohq/sdk": "^0.6.2",
+    "@aleohq/wasm": "^0.6.2",
```
Comment (onetrickwolf) [inline create-aleo-app/template-react-typescript/package.json]: wasm isn't needed in package.json it's included in the SDK

File: create-aleo-app/template-react-typescript/package.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "@types/babel__generator": "^7.6.6",
```
Comment (onetrickwolf) [inline create-aleo-app/template-react-typescript/package.json]: these types should probably be in dev dependencies

File: create-aleo-app/template-react-typescript/package.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    "@types/node": "^20.8.7",
```
Comment (onetrickwolf) [inline create-aleo-app/template-react-typescript/package.json]: same here

File: create-aleo-app/template-react-typescript/.eslintrc.cjs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
```
Comment (onetrickwolf) [inline create-aleo-app/template-react-typescript/.eslintrc.cjs]: would maybe configure this a little more for typescript but fine for now we can iterate

File: create-aleo-app/template-react-typescript/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+  "include": ["build.config.ts", "src", "__tests__", "src/custom.d.ts"],
```
Comment (onetrickwolf) [inline create-aleo-app/template-react-typescript/tsconfig.json]: don't think there's a build.config.ts

**Review by onetrickwolf — CHANGES_REQUESTED** [review]
Looks good I think there's some stuff we could iterate on but I am fine to merge this and iterate so it doesn't get too behind. I would just mainly rename this as `template-react-ts` and make it a variant instead of it's own category.

**Comment by iamalwaysuncomfortable** [comment]
@CalebCurry is there any plan to finish this template?

**Review by onetrickwolf — APPROVED** [review]

**Pull request merged**